gammapy/image/core.py

```diff
@@ -444,27 +447,124 @@ def pad(self, mode, pad_to_factor=None, pad_width=None, **kwargs):
 
-        # converting from unicode to ascii string as a workaround
-        # for https://github.com/numpy/numpy/issues/7112
-        mode = str(mode)
+        if shape is not None:
+            xdiff = shape[1] - self.data.shape[1]
+            ydiff = shape[0] - self.data.shape[0]
+
+            if (np.array([xdiff, ydiff]) % 2).any():
+                raise ValueError('For symmetric padding, difference to new shape '
+                                 'must be even in all axes.')
+
+            xpad, ypad = xdiff // 2, ydiff // 2
+            pad_width = ((ypad, ypad), (xpad, xpad))
+            wcs.wcs.crpix += np.array([xpad, ypad])
 
         data = np.pad(self.data, pad_width=pad_width, mode=mode, **kwargs)
+        return SkyImage(data=data, wcs=wcs)
+
+    def crop(self, shape=None):
+        """
+        Crop sky image symmetrically to a given shape.
+
+        Paramters
+        ---------
+        shape : tuple
+            Desired shape.
+        """
+        xdiff = (self.data.shape[1] - shape[1])
+        ydiff = (self.data.shape[0] - shape[0])
+
+        if (np.array([xdiff, ydiff]) % 2).any():
+            raise ValueError('For symmetric cropping, difference to new shape '
+                             'must be even in all axes.')
+
+        x_crop = xdiff // 2
+        y_crop = ydiff // 2
+        data = self.data[y_crop:-y_crop, x_crop:-x_crop]
+
+        wcs = self.wcs.deepcopy()
+        wcs.wcs.crpix -= np.array([x_crop, y_crop])
+        return SkyImage(data=data, wcs=wcs)
+
+    def downsample(self, factor, method=np.nansum):
+        """
+        Down sample image by a given factor.
+
+        The image is down sampled using `skimage.measure.block_reduce`. If the
+        shape of the data is not divisible by the down sampling factor, the image
+        must be padded beforehand to the correct shape.
 
-        # We don't have to adjust WCS here, because we only pad on the
-        # right and top, and for this change, the CRPIX doesn't change.
+        Parameters
+        ----------
+        factor : int
+            Down sampling factor, must be power of two.
+        method : np.ufunc (np.nansum), optional
+            Method how to combine the image blocks.
+
+        Returns
+        -------
+        image : `SkyImage`
+            Down sampled image.
+        """
+        from skimage.measure import block_reduce
+
+        shape = self.data.shape
+        factor = int(factor)
+        if not (np.mod(shape, factor) == 0).all():
+            raise ValueError('Data shape {0} is not divisable by {1} in all axes.'
+                             'Pad image prior to downsamling to correct'
+                             ' shape.'.format(shape, factor))
+
+        data = block_reduce(self.data, (factor, factor), method)
+
+        # Adjust WCS
+        wcs = _get_resampled_wcs(self, factor, downsampled=True)
+        return SkyImage(data=data, wcs=wcs)
+
+    def upsample(self, factor, order=3):
```

Comment:
Looks like `order=3` is also the default for `scipy.ndimage.zoom`:
http://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.zoom.html

How about changing this to
```
def upsample(self, factor, **kwargs)
```
and passing `kwargs` along to `scipy.ndimage.zoom`?

Then the caller has more power.